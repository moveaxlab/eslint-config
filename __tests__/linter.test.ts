import { CLIEngine } from 'eslint';
import { resolve, relative } from 'path';

const cli = new CLIEngine({
  useEslintrc: true
});

cli.executeOnFiles(['./example/nice.ts']).results.forEach(result => {
  it(`checks that ${relative(resolve('.'), result.filePath)} passes linting`, () => {
    expect(result.errorCount).toBe(0);
    expect(result.warningCount).toBe(0);
  });
});

cli.executeOnFiles(['./example/ugly.ts']).results.forEach(result => {
  it(`checks that ${relative(resolve('.'), result.filePath)} fails to pass linting`, () => {
    expect(result.errorCount).toBeGreaterThan(0);
    expect(result.warningCount).toBeGreaterThan(0);
    expect(result.messages).toMatchSnapshot();
  });
});
