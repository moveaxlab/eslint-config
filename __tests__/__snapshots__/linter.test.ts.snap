// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks that example/ugly.ts fails to pass linting 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 80,
    "endLine": 2,
    "fix": Object {
      "range": Array [
        0,
        80,
      ],
      "text": "function superLongFunction<
  A extends string,
  B extends string,
  C extends string
",
    },
    "line": 1,
    "message": "Replace \`⏎function·superLongFunction<A·extends·string,·B·extends·string,·C·extends·string\` with \`function·superLongFunction<⏎··A·extends·string,⏎··B·extends·string,⏎··C·extends·string⏎\`",
    "nodeType": null,
    "ruleId": "prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 1,
    "endLine": 27,
    "line": 2,
    "message": "File has too many classes (2). Maximum allowed is 1.",
    "messageId": "maximumExceeded",
    "nodeType": "Program",
    "ruleId": "max-classes-per-file",
    "severity": 2,
  },
  Object {
    "column": 10,
    "endColumn": 27,
    "endLine": 2,
    "line": 2,
    "message": "'superLongFunction' is defined but never used.",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/no-unused-vars",
    "severity": 1,
  },
  Object {
    "column": 7,
    "endColumn": 17,
    "endLine": 6,
    "line": 6,
    "message": "'FirstClass' is defined but never used.",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/no-unused-vars",
    "severity": 1,
  },
  Object {
    "column": 7,
    "endColumn": 18,
    "endLine": 14,
    "line": 14,
    "message": "'SecondClass' is defined but never used.",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/no-unused-vars",
    "severity": 1,
  },
  Object {
    "column": 7,
    "endColumn": 17,
    "endLine": 22,
    "line": 22,
    "message": "'myVar' is assigned a value but never used.",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/no-unused-vars",
    "severity": 1,
  },
  Object {
    "column": 14,
    "endColumn": 17,
    "endLine": 22,
    "line": 22,
    "message": "Unexpected any. Specify a different type.",
    "messageId": "unexpectedAny",
    "nodeType": "TSAnyKeyword",
    "ruleId": "@typescript-eslint/no-explicit-any",
    "severity": 2,
  },
  Object {
    "column": 10,
    "endColumn": 26,
    "endLine": 24,
    "line": 24,
    "message": "'logsButShouldNot' is defined but never used.",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/no-unused-vars",
    "severity": 1,
  },
  Object {
    "column": 3,
    "endColumn": 14,
    "endLine": 25,
    "line": 25,
    "message": "Unexpected console statement.",
    "messageId": "unexpected",
    "nodeType": "MemberExpression",
    "ruleId": "no-console",
    "severity": 1,
  },
]
`;
